lib/logstash/filters/jdbc_static.rb

```diff
@@ -0,0 +1,256 @@
+      @scheduler.cron(@schedule, @loader_runner)
+      @scheduler.join
+    else
+      @loader_runner = Util::SingleLoadRunner.new(
+          @processor.local, @parsed_loaders, @local_db_objects)
+    end
+  end
+
+  def global_lookup_options(options = Hash.new)
+    if @tag_on_failure && !@tag_on_failure.empty? && !options.key?("tag_on_failure")
+      options["tag_on_failure"] = @tag_on_failure
+    end
+    if @tag_on_default_use && !@tag_on_default_use.empty? && !options.key?("tag_on_default_use")
+      options["tag_on_default_use"] = @tag_on_default_use
+    end
+    options["lookup_jdbc_driver_class"] = @lookup_jdbc_driver_class
+    options["lookup_jdbc_driver_library"] = @lookup_jdbc_driver_library
+    options["lookup_jdbc_connection_string"] = @lookup_jdbc_connection_string
+    options
+  end
+
+  def add_plugin_configs(options)
+    if @jdbc_driver_library
+      options["jdbc_driver_library"] = @jdbc_driver_library
+    end
+    if @jdbc_driver_class
+      options["jdbc_driver_class"] = @jdbc_driver_class
+    end
+    if @jdbc_connection_string
+      options["jdbc_connection_string"] = @jdbc_connection_string
+    end
+    if @jdbc_user
+      options["jdbc_user"] = @jdbc_user
+    end
+    if @jdbc_password
+      options["jdbc_password"] = @jdbc_password
+    end
+  end
+end end end
+__END__
+
+  def add_plugin_configs(options)
+    if @jdbc_driver_library && !options.key?("jdbc_driver_library")
+      options["jdbc_driver_library"] = @jdbc_driver_library
+    end
+    if @jdbc_driver_class && !options.key?("jdbc_driver_class")
+      options["jdbc_driver_class"] = @jdbc_driver_class
+    end
+    if @jdbc_connection_string && !options.key?("jdbc_connection_string")
+      options["jdbc_connection_string"] = @jdbc_connection_string
+    end
+    if @jdbc_user && !options.key?("jdbc_user")
+      options["jdbc_user"] = @jdbc_user
+    end
+    if @jdbc_password && !options.key?("jdbc_password")
+      options["jdbc_password"] = @jdbc_password
+    end
+  end
+
+
+VARCHAR java.lang.String  setString updateString
+CHAR  java.lang.String  setString updateString
+LONGVARCHAR java.lang.String  setString updateString
+BIT boolean setBoolean  updateBoolean
+NUMERIC java.math.BigDecimal  setBigDecimal updateBigDecimal
+TINYINT byte  setByte updateByte
+SMALLINT  short setShort  updateShort
+INTEGER int setInt  updateInt
+BIGINT  long  setLong updateLong
+REAL  float setFloat  updateFloat
+FLOAT float setFloat  updateFloat
+DOUBLE  double  setDouble updateDouble
+VARBINARY byte[ ] setBytes  updateBytes
+BINARY  byte[ ] setBytes  updateBytes
+DATE  java.sql.Date setDate updateDate
+TIME  java.sql.Time setTime updateTime
+TIMESTAMP java.sql.Timestamp  setTimestamp  updateTimestamp
+CLOB  java.sql.Clob setClob updateClob
+BLOB  java.sql.Blob setBlob updateBlob
+ARRAY java.sql.Array  setARRAY  updateARRAY
+REF java.sql.Ref  SetRef  updateRef
+STRUCT  java.sql.Struct SetStruct updateStruct
```

Comment:
These Derby datatypes need to get into the docs somehow, I have not decided where and what form yet.